lib/Runtime/Library/LazyJSONString.h

```diff
@@ -0,0 +1,89 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#pragma once
+
+namespace Js
+{
+
+struct JSONProperty;
+struct JSONArray;
+
+enum class JSONContentType : uint8
+{
+    Array,
+    Object,
+    Undefined,
+    Null,
+    True,
+    False,
+    Number,
+    String
+};
+
+typedef DListCounted<JSONProperty, Recycler> JSONObject;
+
+struct JSONNumberData
+{
+    Field(Var) value;
+    Field(JavascriptString*) string;
+};
+
+struct JSONProperty
+{
+    Field(JSONContentType) type;
+    Field(const PropertyRecord*) propertyRecord;
```

Comment:
Do we need both the property record and property name? Can we live with just one? By my rough count, this structure is at 17 bytes which is unfortunate since it'll get allocated in the 32 byte bucket, and the wasted space can add up

Reply:
You're right I can probably remove the PropertyRecord case without any real impact. I was overoptimizing for the Parse scenario.